packages/cw721/src/traits.rs

```diff
@@ -1,168 +1,64 @@
-use serde::de::DeserializeOwned;
-use serde::Serialize;
-
-use crate::{
-    AllNftInfoResponse, ApprovalResponse, ApprovalsResponse, ContractInfoResponse, NftInfoResponse,
-    NumTokensResponse, OperatorResponse, OperatorsResponse, OwnerOfResponse, TokensResponse,
-};
-use cosmwasm_std::{Binary, CustomMsg, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
-use cw_utils::Expiration;
-
-pub trait Cw721<T, C>: Cw721Execute<T, C> + Cw721Query<T>
-where
-    T: Serialize + DeserializeOwned + Clone,
-    C: CustomMsg,
-{
-}
-
-pub trait Cw721Execute<T, C>
-where
-    T: Serialize + DeserializeOwned + Clone,
-    C: CustomMsg,
-{
-    type Err: ToString;
-
-    fn transfer_nft(
+use std::fmt::Debug;
+
+use cosmwasm_std::{Deps, Empty, Env, MessageInfo};
+use schemars::JsonSchema;
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::error::Cw721ContractError;
+
+/// This is an exact copy of `CustomMsg`, since implementing a trait for a type from another crate is not possible.
+///
+/// Possible:
+/// `impl<T> Cw721CustomMsg for Option<T> where T: Cw721CustomMsg {}`
+///
+/// Not possible:
+/// `impl<T> CustomMsg for Option<T> where T: CustomMsg {}`
+///
+/// This will be removed once the `CustomMsg` trait is moved to the `cosmwasm_std` crate: https://github.com/CosmWasm/cosmwasm/issues/2056
+pub trait Cw721CustomMsg: Serialize + Clone + Debug + PartialEq + JsonSchema {}
+
+pub trait Cw721State: Serialize + DeserializeOwned + Clone + Debug {}
+
+impl Cw721State for Empty {}
+impl<T> Cw721State for Option<T> where T: Cw721State {}
+impl Cw721CustomMsg for Empty {}
+impl<T> Cw721CustomMsg for Option<T> where T: Cw721CustomMsg {}
```

Comment:
@webmaster128  referring to this ticket https://github.com/CosmWasm/cosmwasm/issues/2056
This is what I had to do as a workaround (line 25) since `impl<T> CustomMsg for Option<T> where T: CustomMsg {}` is missing.